modules/compress/src/test/java/org/apache/ignite/internal/processors/cache/transform/CacheObjectsCompressionTest.java

```diff
@@ -0,0 +1,130 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache.transform;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.binary.BinaryObjectBuilder;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static org.apache.ignite.internal.processors.cache.transform.AbstractCacheObjectsCompressionTest.CompressionTransformer.CompressionType;
+
+/**
+ *
+ */
+@RunWith(Parameterized.class)
+public class CacheObjectsCompressionTest extends AbstractCacheObjectsCompressionTest {
+    /** Thin client. */
+    @Parameterized.Parameter
+    public CompressionType type;
+
+    /** @return Test parameters. */
+    @Parameterized.Parameters(name = "type={0}")
+    public static Collection<?> parameters() {
+        List<Object[]> res = new ArrayList<>();
+
+        for (CompressionType type : CompressionType.values())
+            res.add(new Object[] {type});
+
+        return res;
+    }
+
+    /**
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testCompression() throws Exception {
+        try {
+            CompressionTransformer.type = type;
+
+            Ignite ignite = prepareCluster();
+
+            int i = 42;
+
+            putAndCheck(i, false); // No chances to compress integer.
+
+            String str = "Test string";
+
+            putAndCheck(str, false); // Too short string.
+
+            StringData sd = new StringData(str);
+
+            putAndCheck(sd, false); // Too short wrapped string.
+
+            List<Object> sdList = Collections.singletonList(sd);
+
+            putAndCheck(sdList, false); // Too short wrapped string.
+
+            StringBuilder sb = new StringBuilder();
+
+            for (int k = 0; k < 100; k++)
+                sb.append("AAAAAAAAAA");
+
+            String str2 = sb.toString();
+
+            putAndCheck(str2, type != CompressionType.DISABLED);
+
+            List<Object> list = new ArrayList<>();
+
+            list.add(new BinarizableData(str, null, i));
+
+            putAndCheck(list, false); // Too short list.
+
+            // Adding more elements.
+            list.add(new BinarizableData(str, null, i));
+            list.add(new BinarizableData(str, null, i));
+
+            putAndCheck(list, type != CompressionType.DISABLED); // Enough to be compressed.
```

Comment:
As far as I know we can treat `str` as `handle` during binary object marshalling (we don't do it for strings now, but format supports it and it can be done without compatibility issues), in this case test can fail. At least lets create a new string (`new String(str)` instead of `str`) or append `str` with some char to ensure it will not be marshalled as `handle` if our implementation of binary object marshaller will change.